### ISSUE TITLE
Party/Combatant synchronization issues

### ISSUE THREAD

**Issue by quisquous** [ngld/OverlayPlugin#151]
This is an issue to cover more cases from #123 and the discussion in #137.

I got some reports from somebody where in Eureka, folks were joining the party and the party event didn't have the filled in combatant info:
`
[8/5/2020 02:05:51] Info: raidboss: BrowserConsole: {"type":"PartyChanged","party":[{"id":"10612345","name":"","worldId":0,"job":0,"inParty":true},{"id":"10612346","name":"","worldId":0,"job":0,"inParty":true},{"id":"10612347","name":"","worldId":0,"job":0,"inParty":true},{"id":"10612348","name":"Tini Poutini","worldId":57,"job":31,"inParty":true},{"id":"10612349","name":"Potato Chippy","worldId":99,"job":35,"inParty":true}]} (Source: file:///G:/FFXIV/ACT/Cactbot/resources/party.js, Line: 36)
`

(This causes errors in cactbot because job: 0 maps to `None` and there's no role for that.  That's not the root of the problem, but that's why I saw it.)

I was hoping that my workaround from #137 would cover all cases, but apparently joining the party can also cause a race.  I think this means that polling with a timeout is a better approach, or some different approach entirely?

**Comment by ngld** [comment]
I think I'll make the event source poll the party list until all missing combatants are filled but I'll do it relatively slowly (500ms). That shouldn't cause noticable load and would fix the issue. Whenever a previously missing combatant is found, the event will be raised again with the updated list.

Once I find the relevant network packets, I'll probably replace this whole event with network data.

**Comment by ngld** [comment]
I've done some testing; the NetworkAddCombatant lines arrive after the PartyChanged event which means I can't use them as a fallback. The network lines can't be used as a replacement, either, since they're missing the party/alliance/other status. I haven't had time to look into raw network traffic so instead I've implemented the polling behaviour. It seems to work fine so far but I haven't been able to test the leave/disconnect/etc issues, yet. If you want to test the new implementation, use the debug build. I've added debug messages for each party list change and whenever the event is delayed because party members are still missing.
